tests/e2e/cluster_downgrade_test.go

```diff
@@ -47,13 +48,19 @@ func testDowngradeUpgrade(t *testing.T, clusterSize int) {
 		t.Skipf("%q does not exist", lastReleaseBinary)
 	}
 
-	currentVersion := semver.New(version.Version)
-	lastVersion := semver.Version{Major: currentVersion.Major, Minor: currentVersion.Minor - 1}
+	currentVersion, err := getVersionFromBinary(currentEtcdBinary)
```

Comment:
This looks like unrelated changes from other PR. Please rebase.

Reply:
I can split this into another commit, but it was never passing for me locally. Not sure if that works on your box.

Reply:
I expect that after rebase those changes will not be needed/present.

Reply:
I don't understand, the version inconsistency I run into still exists in main:
https://github.com/etcd-io/etcd/blob/main/tests/e2e/cluster_downgrade_test.go#L50-L55

The issue stems from the fact that the logic takes what is assumed to be in the binary, not what is actually reported by it. If you check the diff, it's already against the latest revision of the test.

Reply:
Can you create a separate PR to fix the downgrade test?

Reply:
sure, I'll pull this out of that PR.

Reply:
created it: https://github.com/etcd-io/etcd/pull/14710